code/controllers/subsystem/explosion.dm

```diff
@@ -295,6 +295,8 @@ SUBSYSTEM_DEF(explosions)
 		var/datum/effect_system/explosion/E
 		if(smoke)
 			E = new /datum/effect_system/explosion/smoke
+		else if(delamination)
+			E = new /datum/effect_system/explosion/delamination
 		else
 			E = new
```

Comment:
Rather than have a boolean delamination flag, a smoke flag etc. we should either:
A) Make it so you pass in the type of the effect system you want to instantiate during the explosion.
B) Have an explosion type variable which is either explosion, smoke or delamination and then switch across that.
This will make the system more expandable for different effect types rather than stacking up the parameter amounts.